src/components/mutation/functional.rs

```diff
@@ -18,7 +19,8 @@ pub fn circular_swap<D: 'static>(permutation: &mut [D], indices: &[usize]) {
     }
 }
 
-#[contracts::debug_requires(indices.len() > 1, "swapping less than two indices is not possible")]
+/// Swaps all `indices` in the `permutation` circularly.
```

Comment:
What is the difference between both circular swap operations?

Reply:
They are functionally equivalent but are implemented differently. I benchmarked both briefly and named the slower one (at least on my machine) with a `2` at the end. We could remove it, but I think it's nice to have a reference of what different versions we tried.

Reply:
Then the documentation should (in my opinion) include that there may be differences in speed between the two. Otherwise it seems unnecessary to have two options.